src/phpDocumentor/Descriptor/Builder/AssemblerInterface.php

```diff
@@ -25,7 +28,7 @@ interface AssemblerInterface
     /**
      * Creates a Descriptor from the provided data.
      *
-     * @param mixed $data
+     * @param Element|Tag|Argument $data
```

Comment:
I am surprised this works; there are so many things used :) 

This is a bad pattern, I know.. The descriptors and assemblers could use some refactoring to make them more type strict. Just so you know, after 3.0 launches we want to see if we can make these objects simple value objects in the domain instead of this complicated thing (especially the inheritance is complicated)

Reply:
(just to prevent confusion; no action is wanted. I just wanted to be transparent and express my surprise :) )

Reply:
At work, we have a massive legacy project, the key is progress step by step. Documenting what the code do now is a good progress. It'll be easier to refactor later if we know exactly what's going on :)

PhpStan and Psalm are big help in that by checking the coherence of the whole and making sure we're not adding mistakes :)

Reply:
Created issue https://github.com/phpDocumentor/phpDocumentor/issues/2245 to specifically target this